src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -349,12 +355,146 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
+        platform=PlatformUpdateParameters(
+            os=None
+        ),
+        agent_configuration=AgentProperties(
+            cpu=None
+        ),
+        timeout=None,
+        step=None,
+        trigger=TriggerUpdateParameters(
+            source_triggers=None,
+            base_image_trigger=None
+        ),
+        credentials=get_custom_registry_credentials(
+            cmd,
+            login_server=login_server,
+            username=username,
+            password=password
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_credential_update(cmd,
+                               client,
+                               task_name,
+                               registry_name,
+                               login_server=None,
+                               username=None,
+                               password=None,
+                               resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    if not login_server or not username or not password:
+        raise CLIError("--login-server, --username and --password are required to update a credential")
+
+    TaskUpdateParameters, PlatformUpdateParameters, AgentProperties, TriggerUpdateParameters = cmd.get_models(
+        'TaskUpdateParameters', 'PlatformUpdateParameters', 'AgentProperties', 'TriggerUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        status=None,
+        platform=PlatformUpdateParameters(
+            os=None
+        ),
+        agent_configuration=AgentProperties(
+            cpu=None
+        ),
+        timeout=None,
+        step=None,
+        trigger=TriggerUpdateParameters(
+            source_triggers=None,
+            base_image_trigger=None
+        ),
+        credentials=get_custom_registry_credentials(
+            cmd,
+            login_server=login_server,
+            username=username,
+            password=password
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_credential_remove(cmd,
+                               client,
+                               task_name,
+                               registry_name,
+                               login_server=None,
+                               resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    if not login_server:
+        raise CLIError("--login-server is required to remove a credential")
+
+    TaskUpdateParameters, PlatformUpdateParameters, AgentProperties, TriggerUpdateParameters = cmd.get_models(
+        'TaskUpdateParameters', 'PlatformUpdateParameters', 'AgentProperties', 'TriggerUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        status=None,
```

Comment:
Do we have to manually set all other properties manually to None one by one? What if I miss one property? Will that cause any grief?